diagnostics-app/src/app/app.component.ts

```diff
@@ -8,32 +8,36 @@
  */
 
 import { Component, HostBinding, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { Theme } from 'src/app/core/enums/global.enums';
+import { ThemeService } from './core/services/theme.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css'],
 })
 export class AppComponent implements OnInit {
-  private _theme: string = 'dark';
-  @HostBinding('class') private _cssClass!: string;
-
-  get theme() {
-    return this._theme;
-  }
+  @HostBinding('class') private _cssClass!: Theme;
+  private _themeSubscription!: Subscription;
 
   get cssClass() {
     return this._cssClass;
   }
 
-  constructor() {}
+  constructor(private _themeService: ThemeService) {}
 
   ngOnInit() {
-    this._cssClass = this.theme;
+    this._cssClass = this._themeService.theme;
+    // listen to theme updates from themeService
+    this._themeSubscription = this._themeService.subscribeOnThemeChange(
+      (nextTheme: Theme) => {
+        this._cssClass = nextTheme;
+      }
```

Comment:
2 nits:
1) WDYT about renaming nextTheme -> newTheme? "next" indicates interest in past data points.